text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,333 @@
+  - framework
+  - learning
+  - steering
+  - typescript
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1003 
+project-link:
+---
+
+<!---
+Directions for above:
+
+stage: Leave as is
+start-date: Fill in with today's date, 2032-12-01T00:00:00.000Z
+release-date: Leave as is
+release-versions: Leave as is
+teams: Include only the [team(s)](README.md#relevant-teams) for which this RFC applies
+prs:
+  accepted: Fill this in with the URL for the Proposal RFC PR
+project-link: Leave as is
+-->
+
+<-- Replace "RFC title" with the title of your RFC -->
+# Deprecate `import Ember from 'ember'; 
+
+## Summary
+
+This RFC proprosing deprecating all APIs that have module-based replacements, as described in [RFC #176](https://rfcs.emberjs.com/id/0176-javascript-module-api) as well as other `Ember.*` apis that are no longer needed.
+
+## Motivation
+
+The `import Ember from 'ember';` set of APIs is implementn as a barrel file, and properly optimizing barrel files [is a lot of work, requiring integration with build time tools](https://vercel.com/blog/how-we-optimized-package-imports-in-next-js).
+
+By removing this set of exports, we have an opportunity to shrink some apps (as some APIs are not used), improving the load performance of ember apps -- and we removing all of these gives us a chance to have a better grasp of what we can get rid of permananently.
+
+Many of these APIs already have alternatives, and those will be called out explicitly in the _Transition Path_ below.
+
+## Transition Path
+
+This list is semi-exhaustive, in that it covers _every_ export from 'ember', but may not exhaustivily provide alternatives.
+
+### New Module Needed
+
+APIs for wiring up a test framework (e.g. QUnit, _etc_)
+- `Ember.Test`
+- `Ember.Test.Adapter`
+- `Ember.Test.QUnitAdapter`
+- `Ember.setupForTesting`
+
+These will need to be moved to a module such as `@ember/testing`.
+
+### A way to communicate with the ember-inspector
+
+The inspector will be hit especially hard by the removal of these APIs.
+
+- `Ember.meta`
+- `Ember.VERSION`
+- `Ember._captureRenderTree` 
+- `Ember.instrument`
+- `Ember.subscribe`
+- `Ember.Instrumentation.instrument`
+- `Ember.Instrumentation.subscribe`
+- `Ember.Instrumentation.unsubscribe`
+- `Ember.Instrumentation.reset`
+- `Ember.ViewUtils`
+- `Ember.ViewUtils.getChildViews`
+- `Ember.ViewUtils.getElementView`
+- `Ember.ViewUtils.getRootViews`
+- `Ember.ViewUtils.getViewBounds`
+- `Ember.ViewUtils.getViewBoundingClientRect`
+- `Ember.ViewUtils.getViewClientRects`
+- `Ember.ViewUtils.getViewElement`
+- `Ember.ViewUtils.isSimpleClick`
+- `Ember.ViewUtils.isSerializationFirstNode`
+
+
+Perhaps we can have folks add this to their apps:
+```js
+import { macroCondition, isDevelopingApp, importSync } from '@embroider/macros';
+
+if (macroCondition(isDevelopingApp())) {
+  // maybe this is side-effecting and installs 
+  // some functions on `globalThis` that the inspector could call
+  importSync('@ember/inspector-support');
```

Comment:
I do not think this is necessary? Inspector can use amd nodules.

Reply:
we will not always be compliing to AMD, so we need something a bit more robust / format-agnostic.

Like, in my side-projects (limber/tutorial on glimdown.com, and others), I want to be shipping actual ESM as my bundled format (some things need to land on _The Platform_ first (multiple modules in one file): 
https://github.com/tc39/proposal-module-declarations

Reply:
my proposal for inspector: 
https://github.com/emberjs/ember.js/pull/20580

Reply:
@patricklx Nice. Just curious, would https://github.com/emberjs/ember.js/pull/20580 also solve https://github.com/emberjs/ember.js/issues/19705? (`Ember.library.register` is still mentioned as the preferred method to register a library in the docs.)

Reply:
We could move the singleton into ember/inspector-support and have apps register through there.
Or use alternatives like unplugin-info and have another api in @ember/inspector-support to register that info